tasks/export_abi.js

```diff
@@ -48,10 +48,14 @@ subtask(
 
     if (!abi.length) return;
 
-    abi = abi.filter((element, index, array) => config.filter(element, index, array, fullName));
-
-    if (config.pretty) {
+    if (config.format == "json") {
+      abi = abi.filter((element, index, array) => config.filter(element, index, array, fullName));
+    } else if (config.format == "minimal") {
       abi = new Interface(abi).format(FormatTypes.minimal);
+    } else if (config.format == "fullName") {
+      abi = new Interface(abi).format(FormatTypes.fullName);
+    } else {
+      throw new HardhatPluginError(`Unknown format: ${config.format}`);
```

Comment:
The `filter` needs to be applied before the conditional block.  The case where `format == "json"` results in no action, but needs to be checked to avoid throwing the "unknown format" error if the user manually specifies "json".

```suggestion
    abi = abi.filter((element, index, array) => config.filter(element, index, array, fullName));

    if (config.format == "minimal") {
      abi = new Interface(abi).format(FormatTypes.minimal);
    } else if (config.format == "fullName") {
      abi = new Interface(abi).format(FormatTypes.fullName);
    } else if (config.format != "json") {
      throw new HardhatPluginError(`Unknown format: ${config.format}`);
```